File: Top 100 Codes/03-Codes for Recursion/02-Prime Number/PrimeNumber.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPrime(int n) {
    if (n < 2) 
        return false; // Handle cases for 0 and 1
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0)
            return false;
    }
    return true;
}

int main() {
    int n;
    
    // Taking user input
    cout << "Enter a number: ";
    cin >> n;

    // Checking and displaying result using if-else
    if (isPrime(n))
        cout << n << " is a Prime Number" << endl;
    else
        cout << n << " is Not a Prime Number" << endl;

    return 0;
}

File: Top 100 Codes/03-Codes for Recursion/02-Prime Number/PrimeNumber.py
import math

def is_prime(n):
    if n < 2:  # Handle cases for 0 and 1
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

# Taking user input
n = int(input("Enter a number: "))

# Checking and displaying result using if-else
if is_prime(n):
    print(f"{n} is a Prime Number")
else:
    print(f"{n} is Not a Prime Number")
